plugins/strategy/action_test.go

```diff
@@ -199,3 +217,194 @@ func TestAction_pushReason(t *testing.T) {
 		})
 	}
 }
+
+func TestPreemptAction(t *testing.T) {
+	testCases := []struct {
+		name     string
+		a        *Action
+		b        *Action
+		expected *Action
+	}{
+		{
+			name: "none vs none",
+			a: &Action{
+				Direction: ScaleDirectionNone,
+			},
+			b: &Action{
+				Direction: ScaleDirectionNone,
+			},
+			expected: &Action{
+				Direction: ScaleDirectionNone,
+			},
+		},
+		{
+			name: "none vs down",
+			a: &Action{
+				Direction: ScaleDirectionNone,
+			},
+			b: &Action{
+				Direction: ScaleDirectionDown,
+			},
+			expected: &Action{
+				Direction: ScaleDirectionDown,
+			},
```

Comment:
In a none vs. down I think the none should win as none ensures greater resource availability. Thoughts?

Reply:
Yeah, that is the safe operation. So we will only scale down if all checks return `ScaleDirectionDown`